build_sysext

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# Copyright (c) 2023 Flatcar Linux. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file. Script to generate sysext. See systemd-sysext(8).
+# Prerequisite that you've run build_packages and build_image
+
+SCRIPT_ROOT=$(dirname "$(readlink -f "$0")")
+. "${SCRIPT_ROOT}/common.sh" || exit 1
+
+# Script must run inside the chroot
+assert_inside_chroot
+assert_root_user
+
+DEFINE_string board "${DEFAULT_BOARD}" \
+  "The board to build an image for."
+
+FLAGS_HELP="USAGE: build_sysext [flags] [package to build].
+This script is used to build a Flatcar sysext. 
+
+Examples:
+
+build_sysext --board=amd64 python - builds python sysext for amd64 architecture.
+
+"
+
+show_help_if_requested "$@"
+
+# Parse command line
+FLAGS "$@" || exit 1
+
+eval set -- "${FLAGS_ARGV:-prod}"
+
+. "${BUILD_LIBRARY_DIR}/release_util.sh" || exit 1
+
+. "${BUILD_LIBRARY_DIR}/toolchain_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/board_options.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/build_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/prod_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/dev_container_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/test_image_content.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/torcx_manifest.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/vm_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/cros_vm_constants.sh" || exit 1
+
+BUILD_DIR="/mnt/host/source/src/build/images/amd64-usr/developer-3621.0.0+nightly-20230531-2100-66-gf786da53f9-a1"
+
+echo "${BUILD_DIR}"
+
+cleanup() {
+  rm -rf "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || true
+  umount "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || true
+  rm "${BUILD_DIR}/${SYSEXTNAME}.raw" || true
+}
+
+# Set up trap to execute cleanup() on script exit
+
+# trap cleanup EXIT
+
+export SYSEXTNAME=$1
+
+case $BOARD in
+*arm64-usr*) ARCH=arm64 ;;
+*amd64-usr*) ARCH=x86-64 ;;
+*) echo "Unknown board" ;;
+esac
+
+source <(grep -P '^(VERSION|VERSION_ID)=' /etc/os-release)
+
+cleanup
+
+mkdir "${BUILD_DIR}/fs-root"
```

Comment:
In what directory is this script supposed to be run? Can we add checks and docs for that (and maybe let the script switch to the right folder by itself)?

Reply:
Ok, if it has no assumptions that's good. Maybe it can't be run under the mounted `/usr` image during image build because of space reasons but I would say that the recommended folder should be the image build folder and document that.